Realm/Realm.Unity/Editor/UnityWeaver.cs

```diff
@@ -103,35 +105,64 @@ private static async Task WeaveAssembly(string assemblyPath)
                 timer.Start();
 
-                var (moduleDefinition, fileStream) = WeaverAssemblyResolver.Resolve(assembly);
-                if (moduleDefinition == null)
+                var resolutionResult = WeaverAssemblyResolver.Resolve(assemblyPath, references);
+                if (resolutionResult == null)
                 {
                     return;
                 }
-                using (fileStream)
-                using (moduleDefinition)
+
+                using (resolutionResult)
                 {
                     // Unity doesn't add the [TargetFramework] attribute when compiling the assembly. However, it's
                     // using NETStandard2, so we just hardcode this.
-                    var weaver = new Weaver(moduleDefinition, logger, new FrameworkName(".NETStandard,Version=v2.0"));
+                    var weaver = new Weaver(resolutionResult.Module, logger, new FrameworkName(".NETStandard,Version=v2.0"));
                     var results = weaver.Execute();
 
-                    moduleDefinition.Write(WriterParameters);
-
                     // Unity creates an entry in the build console for each item, so let's not pollute it.
                     if (results.SkipReason == null)
                     {
+                        resolutionResult.SaveModuleUpdates();
                         logger.Info($"[{name}] Weaving completed in {timer.ElapsedMilliseconds} ms.{Environment.NewLine}{results}");
                     }
                 }
-
-                // save any changes to our weavedAssembly objects
-                // TODO: verify we need to do this
-                AssetDatabase.SaveAssets();
             }
             catch (Exception ex)
             {
-                logger.Warning($"[{name}] Weaving failed: {ex.Message}");
+                logger.Warning($"[{name}] Weaving failed: {ex.StackTrace}");
+            }
+        }
+
+        public void OnPostBuildPlayerScriptDLLs(BuildReport report)
+        {
+            // This is a bit hacky - we need actual references, not directories, containing references, so we pass folder/dummy.dll
+            // knowing that dummy.dll will be stripped.
+            var systemAssemblies = CompilationPipeline.GetSystemAssemblyDirectories(ApiCompatibilityLevel.NET_Standard_2_0).Select(d => Path.Combine(d, "dummy.dll"));
+            var referencePaths = systemAssemblies
+                .Concat(report.files.Select(f => f.path))
+                .ToArray();
+
+            var assembliesToWeave = report.files.Where(f => f.role == "ManagedLibrary");
+            foreach (var file in assembliesToWeave)
+            {
+                WeaveAssemblyCore(file.path, referencePaths);
+            }
+
+            if (report.summary.platform == BuildTarget.iOS)
+            {
+                var realmAssemblyPath = report.files
+                    .SingleOrDefault(r => "Realm.dll".Equals(Path.GetFileName(r.path), StringComparison.OrdinalIgnoreCase))
```

Comment:
Out of curiosity, did you do this for performance optimization or did you actually thought about issues in paths from locale with non-english characters (even danish vowels for example)?

Reply:
It was mostly because I was overly cautious about potential casing differences. I would be extremely surprised if it ever happens, but wanted to cover a case where the build renames Realm.dll to realm.dll. Again, I don't think it will ever happen and we're only looking at the filename, which shouldn't include non-ASCII characters, so it's probably not needed. That being said, it satisfies my OCD, so I'd keep it 😄